sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -523,6 +523,23 @@ int64_t CurlSession::Read(Azure::Core::Context const& context, uint8_t* buffer,
   totalRead = ReadFromSocket(buffer, static_cast<size_t>(readRequestLength));
   this->m_sessionTotalRead += totalRead;
 
+  // Reading 0 bytes means closed connection.
+  // For known content length and chunked response, this means there is nothing else to read from
+  // server or lost connection before getting full response.
+  // For unknown response size, it means the end of response and it's fine.
+  if (totalRead == 0 && (this->m_contentLength > 0 || this->m_isChunkedResponseType))
+  {
+    auto expectedToRead = this->m_isChunkedResponseType ? this->m_chunkSize : this->m_contentLength;
+    if (this->m_sessionTotalRead < expectedToRead)
+    {
+      throw std::runtime_error(
```

Comment:
Why not throw `TransportException` or something inherit from `std::runtime_error`

Reply:
maybe stream exception?
This is while reading with stream. Transport policy could have been done by the time this error happens

Reply:
ReliableStream is catching runtime_error, so, this should be OK to return Trasport Exception. It is also catched by retry policy